.github/workflows/book.yml

```
@@ -55,7 +55,7 @@ jobs:
 
       - name: Upload book artifact
         if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/next' }}
-        uses: actions/upload-pages-artifact@v3
+        uses: actions/upload-pages-artifact@v4
```

```suggestion
        uses: actions/deploy-pages@v4
```

According to the [article](https://github.blog/changelog/2024-12-05-deprecation-notice-github-pages-actions-to-require-artifacts-actions-v4-on-github-com/#what-you-need-to-do)